pkg/test/framework/core/image/settings.go

```diff
@@ -40,6 +43,9 @@ type Settings struct {
 
 	// Image pull policy to use for deployments. If not specified, the defaults of each deployment will be used.
 	PullPolicy string
+
+	// DockerHub value to use in Helm templates for external iamges
+	DockerHub string
```

Comment:
I see why you need this, but I am a bit worried about calling this specifically as DockerHub. That associates the environment variable with the notion of "overriding where the docker.io" is, which is not strictly true. 

Here are a couple of alternative approaches that I can think of:
- Rename the environment variable/flag, so that it is associated with Redis (i.e. `istio.test.redis.hub`). This may not compose well in general. We have issues with component level environment variables/flags causing issues when that particular component is not used in a particular test.
- Add a general boolean flag (e.g. `istio.test.forcehub`?) that indicates that `HUB` should be used in all sub-components as the place to pull images from, as opposed to from whatever their defaults are. I assume there will be only one registry in an air-gapped scenario, so this should be a simple solution that should just work. Though a better name for the flag is certainly warranted.
- Another solution is to force use of `HUB` everywhere (aligned with the single-HUB-in-airgapped-situations reasoning above). This would require us to pull such images to the local registry before starting the tests, which can be done as part of test setup in Prow (in our test scripts). For local execution, we the components can have a setup step that pulls the image first if `istio.test.ci` flag is not set. This is a lot more to implement, but pushes the framework and components to be more hermetic, which is overall goodness IMO.

@nmittler @geeknoid

Reply:
>  pulls the image first if istio.test.ci flag is not set.

I use this flag locally since it gives much more useful logging. Just a data point, we don't need to design it around my workflow

But either way, how does it pull the image? Your cluster could be running on GKE and the test runs locally?

My opinion is just rename the dockerhub to indicate its the hub to use for external dependencies. If that is only redis it could just be redishub I guess.

Reply:
It is about pushing images (i.e.` docker.io/..mumble...redis...`) into `${HUB}`. Every component can do docker pull/push equivalent. (which can be something that the framework natively supports.

Reply:
I don't think we should be pushing to $HUB, I would imagine many people use $HUB where they don't have write access (or they do, but shouldn't be pushing there - I am sure people will accidentally push to official istio gcr if we do this). It seems this will go against the air-gapping, as now we need to push an image instead of just pulling it (pull can be a nop operation if it is already present). I also don't think that would work in kind, as it doesn't use a registry at all

Reply:
The Kubernetes test infrastructure team had the same challenges: https://github.com/kubernetes/kubernetes/blob/master/test/utils/image/manifest.go#L65 the main reason I added a second option to the current available option was that I needed to set `HUB` as `myhub/istio` and `DOCKER_HUB` as `myhub`. From my point of view another possible solution would be to use only `HUB` and adjust the rendered manifests to use `{{ .HUB }}/istio/...` otherwise I would probably go with option 1.

Reply:
@ozevren I am a bit fuzzy on the details, but kind runs kubernetes in a docker container. It then "loads" the images into the container. Then when kubernetes needs the images, they are already present so it doesn't need to pull them.

Reply:
Please do not use `Docker` in any code or documentation.  This is not permissible by their trademark rules.

Reply:
I updated the code to make use of the existing HUB variable